imagesearch/model.py

```diff
@@ -0,0 +1,151 @@
+        kernel_size=3,
+        strides=2,
+        filters=16,
+        padding='SAME',
+        activation=tf.nn.relu) # 128
+    layer_2 = tf.layers.conv2d(
+        inputs=layer_1,
+        kernel_size=3,
+        strides=2,
+        filters=32,
+        padding='SAME',
+        activation=tf.nn.relu) # 64
+    layer_3 = tf.layers.conv2d(
+        inputs=layer_2,
+        kernel_size=3,
+        strides=2,
+        filters=64,
+        padding='SAME',
+        activation=tf.nn.relu) # 32
+    layer_4 = tf.layers.conv2d(
+        inputs=layer_3,
+        kernel_size=3,
+        strides=2,
+        filters=128,
+        padding='SAME',
+        activation=tf.nn.relu) # 16
+    layer_5 = tf.layers.conv2d(
+        inputs=layer_4,
+        kernel_size=3,
+        strides=2,
+        filters=256,
+        padding='SAME',
+        activation=tf.nn.relu) # 8
+    layer_6 = tf.layers.conv2d(
+        inputs=layer_5,
+        kernel_size=3,
+        strides=2,
+        filters=512,
+        padding='SAME',
+        activation=tf.nn.relu) # 4
+    return layer_6
+
+def decoder(inputs):
+    layer_1 = tf.layers.conv2d_transpose(
+        inputs=inputs,
+        kernel_size=3,
+        strides=2,
+        filters=512,
+        padding='SAME',
+        activation=tf.nn.relu) # 8
+    layer_2 = tf.layers.conv2d_transpose(
+        inputs=layer_1,
+        kernel_size=3,
+        strides=2,
+        filters=256,
+        padding='SAME',
+        activation=tf.nn.relu) # 16
+    layer_3 = tf.layers.conv2d_transpose(
+        inputs=layer_2,
+        kernel_size=3,
+        strides=2,
+        filters=128,
+        padding='SAME',
+        activation=tf.nn.relu) # 32
+    layer_4 = tf.layers.conv2d_transpose(
+        inputs=layer_3,
+        kernel_size=3,
+        strides=2,
+        filters=64,
+        padding='SAME',
+        activation=tf.nn.relu) # 64
+    layer_5 = tf.layers.conv2d_transpose(
+        inputs=layer_4,
+        kernel_size=3,
+        strides=2,
+        filters=32,
+        padding='SAME',
+        activation=tf.nn.relu) # 128
+    layer_6 = tf.layers.conv2d_transpose(
+        inputs=layer_5,
+        kernel_size=3,
+        strides=2,
+        filters=3,
+        padding='SAME',
+        activation=tf.tanh) # 256
+    return layer_6
+
+def get_loss(decoded_image, labels, mode):
+    "Return the loss function which will be used with an optimizer."
+
+    loss = None
+    if mode == tf.contrib.learn.ModeKeys.INFER:
+        return loss
+
+    loss = tf.losses.mean_squared_error(decoded_image, labels["image"])
+    return loss
+
+def get_train_op(loss, mode):
+    "Return the trainining operation which will be used to train the model."
+
+    train_op = None
+    if mode != tf.contrib.learn.ModeKeys.TRAIN:
+        return train_op
+
+    global_step = tf.contrib.framework.get_or_create_global_step()
+
+    train_op = tf.contrib.layers.optimize_loss(
+        loss=loss,
+        global_step=global_step,
+        learning_rate=0.01,
```

Comment:
You're right. We weren't using the arg, so I removed it in the other files. I could update it instead to use the arg and I'll switch it back around. Let me know what you think would be best.